test/index.js

```diff
@@ -9,7 +9,9 @@ test('Tests run', (assert) => {
   assert.end();
 });
 
-test('Greet World', (assert) => {
+
+test('Greet World', (assert) => new Promise((resolve) => {
```

Comment:
This ES6 syntax DOES return the promise. Would you like to disable the bug for now and file an issue with ESLint?